src/Glpi/Form/AccessControl/FormAccessControlManager.php

```diff
@@ -92,6 +92,22 @@ public function getActiveAccessControlsForForm(
         return array_values($controls);
     }
 
+    public function isAnonymousForm(Form $form): bool
+    {
+        $access_controls = $this->getActiveAccessControlsForForm($form);
+        return array_reduce(
+            $access_controls,
+            function ($acc, $control) {
+                if ($control->getStrategy() instanceof DirectAccess) {
+                    return $acc || $control->getConfig()->allowUnauthenticated();
+                }
+
+                return $acc;
+            },
+            false
+        );
```

Comment:
This works but is specific to the `DirectAccess` policy (we are planning to allow plugins to add their own policies).

Maybe the ControlType interface should define a method that indicate if a control type allow unauthenticated users ?
There was one at some point but @cedric-anne asked me to delete it and keep all the logic in the `canAnswer` method.

We can re-add it I guess but it must be clear that it is only used to compute warnings for the form editor and is not actually used to compute the form access (the canAnswer method being the only real method that check the session when displaying the form).

What do you think @cedric-anne ? Not too confusing ?